python/tank/descriptor/io_descriptor/path.py

```diff
@@ -92,6 +92,9 @@ def __init__(self, descriptor_dict, sg_connection, bundle_type):
         #  non-required for finding the code)
         self._version = descriptor_dict.get("version") or "Undefined"
 
+        # Resolve environment variables in the version
+        self._version = os.path.expandvars(self._version)
+
```

Comment:
Why don't you also expanduser?

Reply:
I couldn't think of a scenario where it made sense to use `~` in the version field. I can add it if it makes sense

Reply:
Oh wow. Make total sense. Sorry. I looked at the rest of the file and just made a robot comment.